src/mlpack/methods/matrix_completion/matrix_completion_impl.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file matrix_completion_impl.hpp
+ * @author Stephen Tu
+ *
+ * Implementation of MatrixCompletion class.
+ */
+#ifndef __MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_IMPL_HPP
+#define __MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_IMPL_HPP
+
+namespace mlpack {
+namespace matrix_completion {
+
+MatrixCompletion::MatrixCompletion(const size_t m,
+                                   const size_t n,
+                                   const arma::umat& indices,
+                                   const arma::vec& values,
+                                   const size_t r)
+  : m(m), n(n), indices(indices), values(values),
+    sdp(indices.n_cols, 0, CreateInitialPoint(m, n, r))
+{
+  checkValues();
+  initSdp();
+}
+
+MatrixCompletion::MatrixCompletion(const size_t m,
+                                   const size_t n,
+                                   const arma::umat& indices,
+                                   const arma::vec& values,
+                                   const arma::mat& initialPoint)
+  : m(m), n(n), indices(indices), values(values),
+    sdp(indices.n_cols, 0, initialPoint)
+{
+  checkValues();
+  initSdp();
+}
+
+MatrixCompletion::MatrixCompletion(const size_t m,
+                                   const size_t n,
+                                   const arma::umat& indices,
+                                   const arma::vec& values)
+  : m(m), n(n), indices(indices), values(values),
+    sdp(indices.n_cols, 0, CreateInitialPoint(m, n, DefaultRank(m, n, indices.n_cols)))
+{
+  checkValues();
+  initSdp();
+}
+
+void MatrixCompletion::checkValues()
+{
+  if (indices.n_rows != 2)
+    Log::Fatal << "indices.n_rows != 2" << std::endl;
```

Comment:
Can you make these error messages more descriptive?  Should a user come up against this (say they are using CF or something), they're just going to see

```
[FATAL] indices.n_rows != 2
```

and confusion will reign forever.  Probably `MatrixCompletion::MatrixCompletion(): matrix of indices for constraints does not have two rows.` or something similar will be better.  Alternately, I am beginning to think that `Log::Fatal` calls should produce a backtrace, but I want to let the idea simmer more before opening an issue or doing something about it (getting backtraces is not always easy, and I don't think the code we have in place for `Log::Assert` is very robust).